bats/tests/containers/image-allow-list.bats

```diff
@@ -0,0 +1,193 @@
+setup() {
+    load '../helpers/load'
+    RD_USE_IMAGE_ALLOW_LIST=true
+}
+
+@test 'factory reset' {
+    factory_reset
+}
+
+@test 'start container engine' {
+    start_container_engine --kubernetes.enabled=true --kubernetes.version="$RD_KUBERNETES_PREV_VERSION"
+    wait_for_apiserver "$RD_KUBERNETES_PREV_VERSION"
+    wait_for_container_engine
+}
+
+@test 'update the list of patterns first time' {
+    # Can't use bats run with a heredoc so write to a temp file
+    TEMPFILE=$(mktemp -t RD_TEST_63.XXXXX)
+    cat >"$TEMPFILE" <<EOF
+{
+  "version": 6,
+  "containerEngine": {
+    "allowedImages": {
+      "enabled": true,
+      "patterns": ["nginx", "busybox", "python"]
+    }
+  }
+}
+EOF
+    run rdctl api settings -X PUT --input "$TEMPFILE"
+    assert_success
+    rm -f "$TEMPFILE"
+}
+
+@test 'verify pull nginx succeeds' {
+    run ctrctl pull --quiet nginx
+    assert_success
```

Comment:
You don't need to use `run` if all you want is `assert_success`:

```suggestion
    ctrctl pull --quiet nginx
```

Same for the following tests.

Reply:
yup